spec/requests/orders_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe "Orders", type: :request do
+
+  describe "get orders_path" do
+    it "renders the index view" do
+      FactoryBot.create_list(:order, 10)
+      get orders_path
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe "get order_path" do
+    it "renders the :show template" do
+      order = FactoryBot.create(:order)
+      get order_path(id: order.id)
+      expect(response).to render_template(:show)
+    end
+    it "redirects to the index path if the order id is invalid" do
+      get order_path(id: 5000) #an ID that doesn't exist
+      expect(response).to redirect_to orders_path
+    end
+  end
+
+    describe "get new_order_path" do
+      it "renders the :new template" do
+        order = FactoryBot.create(:order)
+        get new_order_path
+        expect(response).to render_template(:new)
+      end
+    end
+
+    describe "get edit_order_path" do
+      it "renders the :edit template" do
+        order = FactoryBot.create(:order)
+        get edit_order_path(id: order.id)
+        expect(response).to render_template(:edit)
+      end
+    end
+
+
+    describe "post orders_path with valid data" do
+      it "saves a new entry and redirects to the show path for the entry" do
+        customer = FactoryBot.create(:customer)
+        order_attributes = FactoryBot.attributes_for(:order, customer_id: customer.id)
+        expect { post orders_path, params: {order: order_attributes}
+      }.to change(Order, :count)
+        expect(response).to redirect_to order_path(id: Order.last.id)
+      end
+    end
+
+    describe "post orders_path with invalid data" do
+      it "does not save a new entry or redirect" do
+        customer = FactoryBot.create(:customer)
+        order_attributes = FactoryBot.attributes_for(:order, customer_id: customer.id)
+        order_attributes.delete(:product_name)
+        expect { post orders_path, params: {order: order_attributes}
+    }.to_not change(Order, :count)
+        expect(response).to render_template(:new)
+      end
+    end
+
+    describe "put order_path with valid data" do
+      it "updates an entry and redirects to the show path for the order" do
+        order = FactoryBot.create(:order)
+        put order_path(id: order.id),   params: {order: {product_count: 123}}
+        order.reload
+        expect(order.product_count).to eq(123)
+        expect(response).to redirect_to order_path(id: order.id)
+      end
+    end
+
+    describe "put order_path with invalid data" do
+      it "does not update the order record or redirect" do
+        order = FactoryBot.create(:order)
+        put order_path(order.id), params: {order: {product_count: 123}}
+        order.reload
+        expect(order.product_count).not_to eq(123)
+        expect(response).to render_template(:edit) 
+      end
+    end
```

Comment:
This rspec test returned failed responses. I believe the product_count should have false/invalid data for there to be a render instead of redirect. You can always check on the rspec error messages, which give you good clues. This is a common part of testing.